dvc/api/dataset.py

```diff
@@ -0,0 +1,61 @@
+from typing import Literal, TypedDict, Union
+
+
+class DVCXDataset(TypedDict):
+    type: Literal["dvcx"]
+    name: str
+    version: int
+
+
+class DVCDataset(TypedDict):
+    type: Literal["dvc"]
+    url: str
+    path: str
+    sha: str
```

Comment:
Minor, but can we name it `rev` to match the rest of the API?

Reply:
This is not a rev, but actually a `sha`/`commit-hash`.